code/modules/multiz/mobile_ladder.dm

```diff
@@ -0,0 +1,107 @@
+/obj/item/mobile_ladder
+	name = "mobile ladder"
+	desc = "A lightweight deployable ladder, which you can use to move up or down. Or alternatively, you can bash some faces in."
+	icon = 'icons/obj/multiz_items.dmi'
+	icon_state = "mobile_ladder"
+	throw_range = 3
+	force = 10
+	w_class = ITEM_SIZE_LARGE
+	slot_flags = SLOT_BACK
+
+/obj/item/mobile_ladder/proc/place_ladder(atom/A, mob/user)
+	var/above = FALSE
+	var/turf/target
+	if(isopenspace(A))
+		target = GetBelow(A)
+		if (!target || (istype(/turf/space, target) || isopenspace(target)))
+			to_chat(user, SPAN_WARNING("There is nothing below you."))
+			return FALSE
+	else if (istype(A, /turf/simulated/floor) || istype(A, /turf/unsimulated/floor))
+		target = GetAbove(A)
+		if (!target || !isopenspace(target))
+			to_chat(user, SPAN_WARNING("There is something above you. You can't deploy \the [src]!"))
+			return FALSE
+		above = TRUE
+
+	if(above)
+		user.visible_message(
+			SPAN_NOTICE("\The [user] begins deploying \the [src] on \the [A]."),
+			SPAN_NOTICE("You begin to deploy \the [src] on \the [A].")
+		)
+	else
+		user.visible_message(
+			SPAN_NOTICE("\The [user] begins to lower \the [src] into \the [A]."),
+			SPAN_WARNING("You begin to lower \the [src] into \the [A].")
+		)
+
+	if (!handle_action(A, user))
+		return
+
+	var/obj/structure/ladder/mobile/ladder = new (target)
+	transfer_fingerprints_to(ladder)
+	ladder = new(get_turf(A))
+	transfer_fingerprints_to(ladder)
+
+	user.drop_from_inventory(src,get_turf(src))
+	qdel(src)
+
+/obj/item/mobile_ladder/afterattack(atom/A, mob/user,proximity)
+	if (!proximity)
+		return
+
+	place_ladder(A,user)
+
+/obj/item/mobile_ladder/proc/handle_action(atom/A, mob/user)
+	if (!do_after(user, 30, user))
+		return FALSE
+	if (!A || QDELETED(src) || QDELETED(user))
```

Comment:
`user` is already QDELETED-checked in `do_after()`
I wonder if `A` shouldn't be QDELETED-checked as well (and/or checked before the `do_after()`)